frontend/packages/console-shared/src/hooks/useBoundingClientRect.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { useResizeObserver } from './useResizeObserver';
+
+type BoundingClientRect = ClientRect | null;
+
+export const useBoundingClientRect = (targetElement: HTMLElement | null): BoundingClientRect => {
+  const [clientRect, setClientRect] = React.useState<BoundingClientRect>(null);
```

Comment:
Remove the effect and use a function in `useState` as a lazy initializer.
```suggestion
  const [clientRect, setClientRect] = React.useState<BoundingClientRect>(() => targetElement ? targetElement.getBoundingClientRect() : null);
```